pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -354,6 +394,9 @@ func (configgen *ConfigGeneratorImpl) buildSidecarInboundListeners(env model.Env
 func (configgen *ConfigGeneratorImpl) buildSidecarOutboundListeners(env model.Environment, node model.Proxy,
 	proxyInstances []*model.ServiceInstance, services []*model.Service) []*xdsapi.Listener {
 
+	// Sort the services in order of creation.
+	services = sortServicesByCreationTime(services)
```

Comment:
Just want to be sure that it works for services that are not loaded from k8s - i.e. we have some default value for time, can be the timestamp of load.

Reply:
The default is a time of 0. It will work, and I have a test for it.